model/metricset/event.go

```diff
@@ -77,7 +77,23 @@ type Sample struct {
 	Name string
 
 	// Value holds the metric value for single-value metrics.
+	//
+	// If Counts and Values are specified, then Value will be ignored.
 	Value float64
+
+	// Values holds the bucket values for histogram metrics.
+	//
+	// These values must be provided in ascending order.
+	Values []float64
+
+	// Counts holds the bucket counts for histogram metrics.
+	//
+	// These numbers must be positive or zero.
```

Comment:
(Where) do you plan to add checks for the restrictions mentioned in the comments?

Reply:
I wasn't planning on adding any checks beyond what Elasticsearch does, I just added that as a reminder/documentation.

Reply:
In case it's not clear: if/when we add support for histogram metrics coming from agents, we would validate these constraints in modeldecoder. In the immediate future these fields will only be set by the metrics computed by the server, so we'll be in full control of the data.